osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs

```diff
@@ -69,6 +69,21 @@ public OsuDifficultyHitObject(HitObject hitObject, HitObject lastLastObject, Hit
             setDistances(clockRate);
         }
 
+        public double Opacity(double mapTime, bool hidden)
+        {
+            double ms = BaseObject.StartTime - mapTime;
+            if (ms < 0)
+                return 0.0;
+
+            double preemptTime = BaseObject.TimePreempt;
+            double fadeInTime = BaseObject.TimeFadeIn;
```

Comment:
I see you removed the clockRate adjustment from these two in a commit, are you sure it's not necessary? The raw values of these don't change via DT, but `DifficultyHitObject.StartTime` (which is passed as `mapTime` to this method) does.